TracerProviderSdk must log if the name is empty

nachoBonafonte opened the pull request:
TracerProviderSdk must log if the name is empty and use a default name. Fixes #139
Also make TracerProviderSdk() non public, since user should use the get() interface
Added tests for TracerProviderSdk
Also added default resources to EnvVarResources

bryce-b commented on Sources/OpenTelemetrySdk/Trace/TracerSdkProvider.swift:
```diff
@@ -20,19 +20,36 @@ import OpenTelemetryApi
 
     /// Returns a new TracerSdkProvider with default Clock, IdGenerator and Resource.
-    public init(clock: Clock = MillisClock(),
+    init(clock: Clock = MillisClock(),
```
I'm a little concerned about making this private. It will limit the configurability of the `TracerSdkProvider`; or is the intended way to make custom configurations subclassing the `TracerSdkProvider`?

per the spec: 
> Configuration (i.e., SpanProcessors, IdGenerator, SpanLimits and Sampler) MUST be managed solely by the TracerProvider and it MUST provide some way to configure all of them that are implemented in the SDK, at least when creating or initializing it.

nachoBonafonte replied:
Yes, you are true, I was following exactly that spec, understanding that configuration should come from the TracerProvider in the API, but we need configuration in the SDK side, since some of those settigns are on the Sdk. I will revert this line

nachoBonafonte commented:
TracerProviderSdk must log if the name is empty and use a default name. Fixes #139
Added tests for TracerProviderSdk
Also added default resources to EnvVarResources
